### ISSUE TITLE
Failed to build project with golang:1.12.0-stretch

### ISSUE THREAD

**Issue by yiyuanliu** [cortexproject/cortex#1298]
Failed to build with docker image golang:1.12.0-stretch. `loadinternal: cannot find runtime/cgo` is reported four times when execute make command.

```
......
time docker run --rm --tty -i \
	-v /home/lyy/go/src/github.com/cortexproject/cortex/.cache:/go/cache \
	-v /home/lyy/go/src/github.com/cortexproject/cortex/.pkg:/go/pkg \
	-v /home/lyy/go/src/github.com/cortexproject/cortex:/go/src/github.com/cortexproject/cortex \
	quay.io/cortexproject/build-image cmd/query-frontend/query-frontend;
make: Entering directory '/go/src/github.com/cortexproject/cortex'
go build -ldflags "-extldflags \"-static\" -linkmode=external -s -w" -tags netgo -i -o cmd/query-frontend/query-frontend ./cmd/query-frontend
# github.com/cortexproject/cortex/cmd/query-frontend
loadinternal: cannot find runtime/cgo
make: Leaving directory '/go/src/github.com/cortexproject/cortex'
......
time docker run --rm --tty -i \
	-v /home/lyy/go/src/github.com/cortexproject/cortex/.cache:/go/cache \
	-v /home/lyy/go/src/github.com/cortexproject/cortex/.pkg:/go/pkg \
	-v /home/lyy/go/src/github.com/cortexproject/cortex:/go/src/github.com/cortexproject/cortex \
	quay.io/cortexproject/build-image cmd/test-exporter/test-exporter;
make: Entering directory '/go/src/github.com/cortexproject/cortex'
go build -ldflags "-extldflags \"-static\" -linkmode=external -s -w" -tags netgo -i -o cmd/test-exporter/test-exporter ./cmd/test-exporter
# github.com/cortexproject/cortex/cmd/test-exporter
loadinternal: cannot find runtime/cgo
make: Leaving directory '/go/src/github.com/cortexproject/cortex'
......
time docker run --rm --tty -i \
	-v /home/lyy/go/src/github.com/cortexproject/cortex/.cache:/go/cache \
	-v /home/lyy/go/src/github.com/cortexproject/cortex/.pkg:/go/pkg \
	-v /home/lyy/go/src/github.com/cortexproject/cortex:/go/src/github.com/cortexproject/cortex \
	quay.io/cortexproject/build-image cmd/alertmanager/alertmanager;
make: Entering directory '/go/src/github.com/cortexproject/cortex'
go build -ldflags "-extldflags \"-static\" -linkmode=external -s -w" -tags netgo -i -o cmd/alertmanager/alertmanager ./cmd/alertmanager
# github.com/cortexproject/cortex/cmd/alertmanager
loadinternal: cannot find runtime/cgo
make: Leaving directory '/go/src/github.com/cortexproject/cortex'
...
time docker run --rm --tty -i \
	-v /home/lyy/go/src/github.com/cortexproject/cortex/.cache:/go/cache \
	-v /home/lyy/go/src/github.com/cortexproject/cortex/.pkg:/go/pkg \
	-v /home/lyy/go/src/github.com/cortexproject/cortex:/go/src/github.com/cortexproject/cortex \
	quay.io/cortexproject/build-image cmd/distributor/distributor;
make: Entering directory '/go/src/github.com/cortexproject/cortex'
go build -ldflags "-extldflags \"-static\" -linkmode=external -s -w" -tags netgo -i -o cmd/distributor/distributor ./cmd/distributor
# github.com/cortexproject/cortex/cmd/distributor
loadinternal: cannot find runtime/cgo
make: Leaving directory '/go/src/github.com/cortexproject/cortex'
...
```

After `kubet create -f ./k8s`, `alermanager, distributor, query-frondend` report CrashLoopBack with no logs.
```
NAME                              READY   STATUS             RESTARTS   AGE
alertmanager-6556c4cd8b-vd87c     0/1     CrashLoopBackOff   6          8m57s
configs-557bbb4557-bg2j9          1/1     Running            0          8m57s
configs-db-55ff748989-b5l59       1/1     Running            0          8m57s
consul-7b9584dcf-wxqmh            1/1     Running            0          8m57s
distributor-8575ccd568-9zrxd      0/1     CrashLoopBackOff   6          8m57s
dynamodb-5457dc4966-gmzwc         1/1     Running            0          8m57s
ingester-56499d4b9f-mnz4q         1/1     Running            0          8m57s
memcached-59d847df6f-gfwhw        1/1     Running            0          8m57s
nginx-55ccdcb7b-c6jjm             1/1     Running            0          8m57s
querier-cf4b7954c-dh6s7           1/1     Running            0          8m56s
query-frontend-578fcf87cf-6gs98   0/1     CrashLoopBackOff   6          8m56s
retrieval-654dcc446-27vwb         1/1     Running            0          8m56s
ruler-66c46bbbfb-ht222            1/1     Running            0          8m56s
s3-547c4db986-mcxr7               1/1     Running            0          8m55s
table-manager-5ff46756f-758qd     1/1     Running            0          8m55s
```

Binary files `./cmd/alertmanager/alertmanager`, `./cmd/distributor/distributor`, `./cmd/query-frontend/query-frontend`, `./cmd/test-exporter/test-exporter` are not executable. All of them exit with `Trace/breakpoint trap (core dumped)`.

I can successfully make the project when change back to golang:1.10.4-stretch in `./build-image/Dockfile`. Here is my system information.
```
lyy@lyy-pc:~/go/src/github.com/cortexproject/cortex$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.10
Release:	18.10
Codename:	cosmic
lyy@lyy-pc:~/go/src/github.com/cortexproject/cortex$ go version
go version go1.12.1 linux/amd64
lyy@lyy-pc:~/go/src/github.com/cortexproject/cortex$ docker version
Client:
 Version:           18.09.3
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        774a1f4
 Built:             Thu Feb 28 06:47:20 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.3
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       774a1f4
  Built:            Thu Feb 28 05:59:55 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Comment by bboreham** [comment]
Hi, can you make sure you're building from code after #1280 was merged, i.e. after commit 00eea14409bd2ff050d49ce14f12b958a41bdc27

This adds `CGO_ENABLED=0` to the build and changes some flags.

**Comment by yiyuanliu** [comment]
@bboreham I was using code before #1280 merged. This fixed by #1280. Thanks.
